doc/build_antora.sh

```diff
@@ -7,6 +7,21 @@
 # Official repository: https://github.com/boostorg/url
 #
 
+set -e
+
+# If CXX was not already set, then determine the newest clang.
+if [ ! -n "$CXX" ]; then
```

Comment:
The workflow already sets CXX and CC. Is this change to ensure it works in the container that builds the preview?